browser/ui/views/page_action/wayback_machine_state_manager.cc

```diff
@@ -0,0 +1,73 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/page_action/wayback_machine_state_manager.h"
+
+#include "base/functional/bind.h"
+#include "brave/browser/ui/views/page_action/wayback_machine_action_icon_view.h"
+#include "brave/components/brave_wayback_machine/brave_wayback_machine_tab_helper.h"
+#include "brave/components/brave_wayback_machine/wayback_state.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+
+WaybackMachineStateManager::WaybackMachineStateManager(
+    WaybackMachineActionIconView* icon,
+    Browser* browser)
+    : icon_(*icon), browser_(*browser) {
+  CHECK(browser);
+  browser_->tab_strip_model()->AddObserver(this);
+}
+
+WaybackMachineStateManager::~WaybackMachineStateManager() {
+  browser_->tab_strip_model()->RemoveObserver(this);
```

Comment:
I think we didn't need to call RemoveObserver() back then for TabStripModel. Did they changed? Anyway, if we need to call RemoveObserver(), can you using ScopedObserver?

Reply:
ah, always forgot about it. it's removed by itself. deleted.